src/commonMain/kotlin/io/github/detekt/sarif4k/SarifSerializer.kt

```diff
@@ -15,4 +20,54 @@ object SarifSerializer {
     fun toJson(sarif: SarifSchema210): String = json.encodeToString(sarif)
 
     fun fromJson(json: String): SarifSchema210 = Json.decodeFromString(json)
-}
\ No newline at end of file
+}
+
+// Extracted from https://github.com/Kotlin/kotlinx.serialization/issues/296#issuecomment-1859572541
+private fun Any?.toJsonElement(): JsonElement = when (this) {
```

Comment:
Rhetorical question: does this need to be symmetric? If one puts in an IntArray for serialisation and get back a List<Int> on deserialisation, is that acceptable? As far as I understand that's the current behaviour.

Reply:
I think that's acceptable. In general you will never read and write at the same time. Only us do that on the tests.

In general what you want is to build a `SarifSchema210` using any type you want and then serialize it to json. Or you just want to read a Sarif and do something with it. And in that case you want simple structures to avoid a cast nightmare. Using `Any?` is not a good idea in general but that's the only way to implement what Sarif defines.

Reply:
> In general you will never read and write at the same time.

Except when you read multiple files and merge them into one to save the result. I guess in that case it'll be consistent, because the origin and resulting are both serialized string.

The problem could be if one Sarif object is deserialized from File, another Sarif object generated generated, and the two merged, but in that case the "other key wins" rule will probably solve the problem and it won't try to merge List<Int> with IntArray.